src/main/groovy/cruise/umple/UmplePlugin.groovy

```diff
@@ -5,16 +5,29 @@ import cruise.umple.UmpleConsoleConfig
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 
-class UmplePlugin implements Plugin<Project> {
+class UmpleGradlePlugin implements Plugin<Project> {
 
     private UmpleConsoleConfig cfg
 
     @Override
     void apply(final Project project) {
 
-
-        project.task('hello') << {
-            println "Hello, World!"
+        project.task('compileUmpleFile') << {
+			UmpleConsoleMain(cfg)
+		
+			if(project.hasProperty("umpleArgs"))
+			{
+				// arguments are specified through gradle by -P, separated by commas (-P is for project properties)
+				// eg: "gradle compileUmpleFile 'P-umpleArgs=test.ump,-g,Java'"
+				args(umpleArgs.split(','))
```

Comment:
Use these arguments to change the `UmpleConsoleConfig` parameter, it's why I wrote it originally.